src/server/controllers/categories.ts

```diff
@@ -0,0 +1,74 @@
+import { compare, genSalt, hash } from 'bcrypt-nodejs'
+import { NextFunction, Request, Response } from 'express'
+import * as moment from 'moment'
+import { CategoryInterface } from '../interfaces'
+import { logger, redisCache, facebook, googleAuth, sendgrid } from '../services'
+import { Category } from '../models'
+import { ErrorPayload } from '../errorPayload'
+
+export async function loadAsync(request: Request, response: Response, next: NextFunction, id: string) {
+    try {
+        const category = await CategoryInterface.getAsync(id)
+
+        if (!category) { return response.status(404).json(new ErrorPayload(404, 'Category not found')) }
+
+        response.locals.category = category
+        next()
+    } catch (error) {
+        handleError(error, response)
+    }
+}
+
+export async function getOneAsync(request: Request, response: Response) {
+    try {
+        const category: Category = response.locals.category
+        if (!category) { throw new ErrorPayload(404, 'Category not found') }
+
+        const bodyResponse = { category: Category.toDTO(category) }
+        response.status(200).json(bodyResponse)
+    } catch (error) {
+        handleError(error, response)
+    }
+}
+
+export async function getListAsync(request: Request, response: Response) {
+    try {
+        const categories = await CategoryInterface.findAsync({})
+        if (!categories) { throw new ErrorPayload(500, 'Failed to get categories') }
+
+        const bodyResponse = { categories: categories.map((c) => Category.toDTO(c)) }
+        response.status(200).json(bodyResponse)
+    } catch (error) {
+        handleError(error, response)
+    }
+}
+
+export async function createAsync(request: Request, response: Response) {
+    try {
+        const { name, deleted } = request.body
```

Comment:
Deleted should not be received. It should be always setted to false when creating a category